src/app/services/configuration-updater/configurationUpdater.service.ts

```diff
@@ -0,0 +1,341 @@
+      updatedConfig = this.updateMapContributors(updatedConfig, availableFields);
+      updatedConfig = this.updateChipSearchContributors(updatedConfig, availableFields);
+      return updatedConfig;
+  }
+
+  /**
+   * Removes the properties -from map component - including fields that are not available for exploration
+   * @param data configuration object
+   * @param availableFields List of available fields for exploration
+   * @returns configuration object
+   */
+  public updateMapComponent(data, availableFields: Set<string>): any {
+    if (data && data.arlas.web.components) {
+      const mapComponentConfig = data.arlas.web.components.mapgl;
+      if (mapComponentConfig && mapComponentConfig.input) {
+        const idFeatureField = mapComponentConfig.input.idFieldName;
+        if (idFeatureField && !availableFields.has(idFeatureField)) {
+          delete mapComponentConfig.input.idFieldName;
+        }
+      }
+    }
+    return data;
+  }
+
+  /**
+   * Removes the properties -from ResultListContributor- that define fields not available for exploration
+   * @param data configuration object
+   * @param availableFields List of available fields for exploration
+   * @returns configuration object
+   */
+  public updateResultListContributor(data, availableFields: Set<string>): any {
+    data.arlas.web.contributors.filter(contributor => contributor.type === 'resultlist').forEach(contributor => {
+      if (contributor.fieldsConfiguration) {
+        const fc = contributor.fieldsConfiguration;
+        /** remove imageFieldName */
+        if (fc.imageFieldName && !availableFields.has(fc.imageFieldName)) {
+          delete fc.imageFieldName;
+          delete fc.urlImageTemplate;
+        }
+        /** remove thumbnailFieldName */
+        if (fc.thumbnailFieldName && !availableFields.has(fc.thumbnailFieldName)) {
+          delete fc.thumbnailFieldName;
+          delete fc.urlThumbnailTemplate;
+        }
+        /** remove urlImageTemplate */
+        if (fc.urlImageTemplate) {
+          const url = fc.urlImageTemplate;
+          const eventualField = url.substring(
+            url.lastIndexOf('{') + 1,
+            url.lastIndexOf('}')
+          );
+          if (eventualField && eventualField.length > 0 && !availableFields.has(eventualField)) {
+            delete fc.imageEnabled;
+            delete fc.urlImageTemplate;
+          }
+        }
+        /** remove urlThumbnailTemplate */
+        if (fc.urlThumbnailTemplate) {
+          const url = fc.urlThumbnailTemplate;
+          const eventualField = url.substring(
+            url.lastIndexOf('{') + 1,
+            url.lastIndexOf('}')
+          );
+          if (eventualField && eventualField.length > 0 && !availableFields.has(eventualField)) {
+            delete fc.thumbnailEnabled;
+            delete fc.urlThumbnailTemplate;
+          }
+        }
+        /** remove titleFieldNames.fieldPath */
+        if (fc.titleFieldNames) {
```

Comment:
Is ot possible to factorize the deletion in a dedicated method lile `deleteTitleFieldNames(); deleteTooltipFieldNames()` as the deletion does always the same?